dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java

```diff
@@ -0,0 +1,166 @@
+    private static final long serialVersionUID = 1L;
+
+    ELProcessor elp;
+
+    public EL50BasicTestsServlet() {
+        super();
+
+        elp = new ELProcessor();
+        elp.defineBean("simpleBean", new SimpleBean());
+    }
+
+    /**
+     * 
+     * Expression Language 5.0 in Jakarta EE10 added generics to methods in multiple classes https://github.com/jakartaee/expression-language/issues/157
+     * Classes with changes we tests are ELProcessor, ExpressionFactory, ELContext, ValueExpression, ELResolver (returns null)
+     * Classes with changes we don't test or implicitly test TypeConverter(returns null), CompositeELResolver(returns null)
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void testGenerics() throws Exception {
+        // Test generics with ELProcessor uses ValueExpression.getValue under the covers that uses generics
+        // No need to cast this to boolean like previous versions
+        assertTrue("The type was expected to be true for ELProcessor but was not", elp.eval("simpleBean.getNumber() == 25"));
+
+        // Test generics with ExpressionFactory
+        ExpressionFactory factory = ExpressionFactory.newInstance();
+        Integer number = 12;
+        Double result = factory.coerceToType(number, java.lang.Double.class);
+        assertNotNull(result);
+        assertTrue("The type was expected to be coerced to Double for ExpressionFactory but was not: " + result, result.equals(new Double(number)));
+
+        // Test generics with ELContext uses ELResolver.convertToType under the covers
+        ELContext elctx = new StandardELContext(ELManager.getExpressionFactory());
+        elctx.convertToType(number, java.lang.Double.class);
+        assertNotNull(result);
+        assertTrue("The type was expected to be converted to Double for ELContext but was not: " + result, result.equals(new Double(number)));
+
+        // Test generics with ValueExpression technically not necessary since it is used in the covers of elprocessor
+        ValueExpression valueExp = factory.createValueExpression(number, java.lang.Double.class);
+        result = valueExp.getValue(elctx);
+        assertNotNull(result);
+        assertTrue("The type was expected to be Double for ValueExpression but was not: " + result, result.equals(new Double(number)));
+
+        // Test generics with ELResolver used in ELContext implementation but normal elr.convertToType returns null if not implemented
+        elctx = new StandardELContext(ELManager.getExpressionFactory());
+        ELResolver elr = elctx.getELResolver();
+        result = elr.convertToType(elctx, number, java.lang.Double.class);
+        assertNull("The result was expected to be null for ELResolver but was not: " + result, result);
+    }
+
+
+    /**
+     * 
+     * Expression Language 5.0 in Jakarta EE10 switched ELResolver getFeatureDescriptors to return a default value of null and deprecate it to remove in EL 6.0.
```

Comment:
Are we sure it will be removed in 6.0 or should we just say a future version?

Reply:
Yes, at least from the [changes](https://github.com/jakartaee/expression-language/commit/4cfe10f760d99dd68ec43bbe9fa02baa7ae26714#diff-d6f984fc80d47beafbcd3f2b2ae262a7b3f25578e2ce1b9d86b29d3d7c8ad8c6R299) in the open source it's documented to be removed without replacement for EL 6.0